var/spack/repos/builtin/packages/nccmp/package.py

```diff
@@ -28,4 +30,18 @@ def cmake_args(self):
         if cflags:
             args.append(self.define("CMAKE_C_FLAGS", " ".join(cflags)))
 
+        nc = self.spec["netcdf-c"]
+        if "~shared" in nc:
+            nc_pc_cmd = ["nc-config","--static","--libs"]
+            nc_flags = \
+              subprocess.check_output(nc_pc_cmd, encoding="utf8").strip()
+            args.append(self.define("CMAKE_EXE_LINKER_FLAGS", nc_flags))
+            if "+mpi" in nc:
+                args.append(
+                  self.define(
+                    "CMAKE_C_COMPILER",
+                    self.spec["mpi"].mpicc
```

Comment:
I am getting this error on Cheyenne:
```
/glade/work/heinzell/spack-stack-tst-120/spack-stack/spack/var/spack/repos/builtin/packages/nccmp/package.py:40, in cmake_args:
         37            args.append(self.define("CMAKE_EXE_LINKER_FLAGS", nc_flags))
         38
         39        if self.spec.satisfies("^mpi"):
  >>     40            args.append(self.define("CMAKE_C_COMPILER", self.spec["mpi"].mpicc))
         41
         42        return args
```
Again a weird one since I don't see `~shared` in the netcdf-c flags.